addons/stock/views/stock_production_lot_views.xml

```diff
@@ -11,6 +11,11 @@
         <field name="model">stock.production.lot</field>
         <field name="arch" type="xml">
             <form string="Lots/Serial Numbers">
+                <div class="oe_form_box_info oe_text_center oe_edit_only" attrs="{'invisible': [('quant_ids','=',[])]}">
+                    <p><strong>
+                    You cannot change the product of a lot that already contains quants.
```

Comment:
Please change string -> You can no longer change the product of this lot/serial number since it has been already moved.